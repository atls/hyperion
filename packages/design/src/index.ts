export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
