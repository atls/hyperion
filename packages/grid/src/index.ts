export * from './Column'
export * from './Grid'
