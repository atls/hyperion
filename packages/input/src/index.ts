export * from './Textarea'
export * from './Input'
export * from './Checkbox'
