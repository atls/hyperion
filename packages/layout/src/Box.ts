import styled                                                                 from '@emotion/styled'
import { border, boxShadow, color, flexbox, layout, position, space, system } from 'styled-system'

export const Box = styled.div(
  system({
    boxSizing: true,
    overflowX: true,
    overflowY: true,
  }),
  layout,
  space,
  color,
  flexbox,
  boxShadow,
  position,
  border
)

Box.defaultProps = {
  display: 'flex',
  boxSizing: 'border-box',
}
