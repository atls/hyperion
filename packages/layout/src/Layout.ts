import { color, flexbox, layout, position, space, system } from 'styled-system'

import styled                                              from '@emotion/styled'

const boxSizing = system({
  boxSizing: true,
})

export const Column = styled.div(layout, space, flexbox, position, boxSizing)

Column.defaultProps = {
  width: '100%',
  flexDirection: 'column',
  display: 'flex',
}

export const Row = styled.div(layout, space, flexbox, position, boxSizing)

Row.defaultProps = {
  width: '100%',
  flexDirection: 'row',
  display: 'flex',
}

export const Layout = styled.div(layout, space, flexbox, position, color, boxSizing)

Layout.defaultProps = {
  display: 'flex',
}
