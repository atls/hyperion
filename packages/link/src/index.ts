export * from './Link'
export * from './NextLink'
export * from './RouteLink'
