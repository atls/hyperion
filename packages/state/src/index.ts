export * from './withState'
export * from './useActive'
export * from './useHover'
export * from './useFocus'
