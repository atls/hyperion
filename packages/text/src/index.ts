export * from './Text'
export * from './Space'
export * from './SearchResult'
