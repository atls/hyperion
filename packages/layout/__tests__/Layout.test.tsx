import React      from 'react'
import renderer   from 'react-test-renderer'

import { Layout } from '../src/index'

it('Layout should render correctly', () => {
  const tree = renderer.create(<Layout background-color='#000' size='200' />).toJSON()

  expect(tree).toMatchSnapshot()
})
