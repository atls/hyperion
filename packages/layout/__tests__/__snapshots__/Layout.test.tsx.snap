// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
.emotion-0 {
  width: 200;
  height: 200;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-0"
  display="flex"
  size="200"
/>
`;
